### DISCUSSION TITLE
Is it possible to get ImageSharp to tell you the bounds of text it just rendered?

### DISCUSSION THREAD

**Discussion by jez9999** [SixLabors/Fonts#430] [Q&A] [answered]
I'd like to write several text blocks, but I want them to be positioned next to where the other text block was written.  Because I don't know what the length or font size of each text block will be, I need ImageSharp to tell me what the bounds were for the text it just rendered with:

```
imgSlide.Mutate(x =>
	x.DrawText(
		new RichTextOptions(normalFontQnText) {
			Origin = new Vector2(810, 800),
			LayoutMode = LayoutMode.HorizontalTopBottom,
			HorizontalAlignment = HorizontalAlignment.Center,
			VerticalAlignment = VerticalAlignment.Center,
			TextAlignment = TextAlignment.Start,
			WrappingLength = 900,
			TextRuns = parsed.TextRuns,
		},
		parsed.ParsedString,
		Color.Black
	)
);
```

Is it possible to get this information somehow from ImageSharp?

**Comment by tocsoft** [comment, marked as answer]
https://docs.sixlabors.com/articles/fonts/gettingstarted.html the first example for our getting start docs cover this exact sort of case.

```c# 
var size = TextMeasurer.MeasureAdvance(parsed.ParsedString, new RichTextOptions(normalFontQnText) {
			Origin = new Vector2(810, 800),
			LayoutMode = LayoutMode.HorizontalTopBottom,
			HorizontalAlignment = HorizontalAlignment.Center,
			VerticalAlignment = VerticalAlignment.Center,
			TextAlignment = TextAlignment.Start,
			WrappingLength = 900,
			TextRuns = parsed.TextRuns,
		} );
```
is basically what your looking for.

PS this question would have been better asked in [ImageSharp.Drawing](https://github.com/SixLabors/ImageSharp.Drawing) as its a `DrawXXX` API or [Fonts](https://github.com/SixLabors/Fonts/) as its about text metadata.

**Reply by jez9999** [reply]
Thanks.  Interestingly the FontRectangle you get back doesn't seem to take into account the Origin of the RichTextOptions meaning that Height and Bottom are the same, and Location is 0,0.  Not sure whether this is intended?

**Reply by JimBobSquarePants** [reply]
It is. The advance is the size of the glyphs plus any spacing. It’s not a measure of bounds.
